esti/gc_utils_test.go

```diff
@@ -30,25 +32,30 @@ func getDockerArgs(workingDirectory string, localJar string) []string {
 	}
 }
 
-// runCommand runs the given command. It redirects the output of the command:
-// 1. stdout is redirected to the logger.
-// 2. stderr is simply printed out.
-func runCommand(cmdName string, cmd *exec.Cmd) error {
-	handlePipe := func(pipe io.ReadCloser, log func(format string, args ...interface{})) {
-		reader := bufio.NewReader(pipe)
-		go func() {
-			defer func() {
-				_ = pipe.Close()
-			}()
-			for {
-				str, err := reader.ReadString('\n')
-				if err != nil {
-					break
-				}
-				log(strings.TrimSuffix(str, "\n"))
+// handlePipe calls log on each line of pipe, and writes nil or an error to
+// ch when done.
+func handlePipe(pipe io.ReadCloser, log func(messages ...interface{}), ch chan<- error) {
+	reader := bufio.NewReader(pipe)
+	go func() {
+		var err error
+		for {
+			str, err := reader.ReadString('\n')
+			if err != nil {
+				break
 			}
-		}()
-	}
+			log(strings.TrimSuffix(str, "\n"))
+		}
+		if err == io.EOF {
+			err = nil
+		}
+		ch <- err
+		pipe.Close()
```

Comment:
IIUC, this is not needed?

Reply:
You're right, not needed here.

Removed, thanks!